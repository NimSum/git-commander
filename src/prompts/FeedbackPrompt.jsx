import React from 'react';

export function FeedbackPrompt (props) {
  let elContainer;

  switch(true) {
    case props.reset:
      elContainer = 
        (<article className="user-prompt">
          <h3>Incorrect!</h3>
          <p>Would you like to try again ?</p>
          <button 
            onClick={ props.resetGame } 
            type="button">
            PPPPPRACTICE
          </button>
        </article>)
    break;
    case props.winner:
      elContainer = 
        (<article className="user-prompt">
          <h3>You have reached Remote Repo!</h3>
          <p>You can keep going by pressing button below </p>
          <button 
            onClick={ props.nextRound } 
            type="button">
            git Good
          </button>
        </article>)
    break;
    case props.correct === 'incorrect':
    elContainer = 
      (<div className="solution-box">
        <h4>Incorrect!</h4>
        <p>The correct command was:</p>
        <p>{ props.currentQuestion.answer }</p>
      </div>)
    break;
    case props.correct === 'correct':
    elContainer = 
      (<div className="solution-box">
        <h4>Correct!</h4>
        <p>Your answer was:</p>
        <p>{ props.currentQuestion.answer }</p>
      </div>)
    break;
    default: 
    elContainer = (<div>HI</div>)
  }

  return elContainer;
}

export default FeedbackPrompt;
