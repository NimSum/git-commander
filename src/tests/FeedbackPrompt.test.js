import React from 'react';
import FeedbackPrompt from '../prompts/FeedbackPrompt';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Game Window' , () => {
  const mockQuestion =  {
    "statement" : "Our Git spaceship needs to switch to a different branch to steer the ship!",
    "detail" : "Switch to the branch called \"steer-ship\" to avoid the obstacle",
    "answer" : "git checkout steer-ship", 
    "difficulty" : "1",
    "moreInfo" : "/////A link to where they could read more about this command",
    "id": "1"
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      < FeedbackPrompt
        currentQuestion={ mockQuestion }
      />
    );
  });

  it('Should match snapshot of game reset', () => {
    const mockResetProp = { reset: true };
    wrapper.setProps(mockResetProp);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should match snapshot of winning', () => {
    const mockWinnerProp = { winner: true };
    wrapper.setProps(mockWinnerProp);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should match snapshot incorrect answer', () => {
    const mockIncorrectProp = { correct: 'incorrect' };
    wrapper.setProps(mockIncorrectProp);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should match snapshot or correct answer', () => {
    const mockCorrectProp = { correct: 'correct' };
    wrapper.setProps(mockCorrectProp);
    expect(wrapper).toMatchSnapshot();
  })
  
});