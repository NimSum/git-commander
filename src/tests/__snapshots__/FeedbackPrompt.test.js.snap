// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Window Should match snapshot incorrect answer 1`] = `
<div
  className="solution-box"
>
  <h4>
    Incorrect!
  </h4>
  <p>
    The correct command was:
  </p>
  <p>
    git checkout steer-ship
  </p>
</div>
`;

exports[`Game Window Should match snapshot of game reset 1`] = `
<article
  className="user-prompt"
>
  <h3>
    Incorrect!
  </h3>
  <p>
    Would you like to try again ?
  </p>
  <button
    type="button"
  >
    Purractice!
  </button>
</article>
`;

exports[`Game Window Should match snapshot of winning 1`] = `
<article
  className="user-prompt"
>
  <h3>
    You have reached Remote Repo!
  </h3>
  <p>
    You can keep going by pressing button below 
  </p>
  <button
    type="button"
  >
    git Good
  </button>
</article>
`;

exports[`Game Window Should match snapshot or correct answer 1`] = `
<div
  className="solution-box"
>
  <h4>
    Correct!
  </h4>
  <p>
    Your answer was:
  </p>
  <p>
    git checkout steer-ship
  </p>
</div>
`;
